rtlib/rope.hh

```diff
@@ -622,8 +701,26 @@ class Ref {
     }
 };
 
```

Comment:
Programs using the `Rope` type can now be checkpointed as well. 

In case you're wondering why I changed the overload for `operator<<` and the `put` method in the `Rope` class: While trying to write the `serialize` method for the `Rope` class I encountered some really weird behaviour from `boost`, which seemed to completely ignore the instructions I put in the `serialize` method and  just wrote the `Block` content into the archives, even when I didn't even provide a `serialize` method for `Rope`, which would usually throw an error. As it turns out, the overloaded and TEMPLATED `<<` operator of the `Rope` class overwrote `boost`s `<<` operator, which is used to pipe data into an archive. As a result, the `serialize` method I wrote was never actually called, but instead the `put` method was called (which is usually used for printing to `std::cout`) since it also templated and accepts any stream as input and wrote the `Block` content into the archive, which obviously caused some really weird behaviour and ultimately corrupted all of the archives.

To prevent this I added overloads for `operator<<` for all relevant streams and removed the generic template so `boost` gets to use it's own `operator<<` overload. 

This goes to show that one has to be really careful when using generic template arguments, as this can definitely lead to some unforeseen bugs/problems later on.